Do not compute the magic number but use constant value instead,
this fixes integer overflow on big-endian platform
(debian #635370, busybox #4405)

Index: debian/include/archive.h
===================================================================
--- debian.orig/include/archive.h	2011-10-30 22:12:45.256454690 +0400
+++ debian/include/archive.h	2011-10-30 22:31:37.499003659 +0400
@@ -14,7 +14,7 @@
 	XZ_MAGIC1   = 256 * 0xfd + '7',
 	XZ_MAGIC2   = 256 * (256 * (256 * 'z' + 'X') + 'Z') + 0,
 	/* Different form: 32 bits, then 16 bits: */
-	XZ_MAGIC1a  = 256 * (256 * (256 * 0xfd + '7') + 'z') + 'X',
+	XZ_MAGIC1a  = 0xFD377A58UL /* 256 * (256 * (256 * 0xfd + '7') + 'z') + 'X' */,
 	XZ_MAGIC2a  = 256 * 'Z' + 0,
 #else
 	COMPRESS_MAGIC = 0x9d1f,
